File: BAEKJOON/java/3062_수_뒤집기.java
import java.io.*;
import java.util.ArrayList;

public class Main {
    public static String s;

    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        int tc = Integer.parseInt(br.readLine());
        for (int i = 0; i < tc; i++) {
            input();
            solution();
        }

        bw.flush();
        bw.close();
    }

    private static void input() throws IOException {
        s = br.readLine();
    }

    private static void solution() throws IOException {
        StringBuilder revS = new StringBuilder(s);
        revS.reverse();

        int sum = Integer.parseInt(s) + Integer.parseInt(revS.toString());

        String strSum = String.valueOf(sum);
        StringBuilder revStrSum = new StringBuilder(strSum);
        revStrSum.reverse();

        if (strSum.equals(revStrSum.toString())) {
            bw.write("YES\n");
            return;
        }

        bw.write("NO\n");
    }
}

File: BAEKJOON/cpp/3062_수_뒤집기.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

string s, rev_s;

void input() {
	cin >> s;
}

void solution() {
	rev_s = s;
	reverse(rev_s.begin(), rev_s.end());

	int sum = stoi(s) + stoi(rev_s);

	string str_sum = to_string(sum);
	string rev_str_sum = str_sum;
	reverse(rev_str_sum.begin(), rev_str_sum.end());

	if (str_sum == rev_str_sum) {
		cout << "YES" << "\n";
		return;
	}

	cout << "NO" << "\n";
}

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	int tc;
	cin >> tc;

	for (int i = 0; i < tc; i++) {
		input();
		solution();
	}

	return 0;
}